dapp/test/crowdsale.js

```diff
@@ -56,6 +57,44 @@ class TestBosonRunner {
     }
 }
 
+// Get Ether balance of a given account. Why does this not exist in
+// truffle-contract?
+async function getBalance(account) {
+    return new Promise((resolve, reject) => {
+        // Ugh.
+        const web3 = Crowdsale.web3;
+        web3.eth.getBalance(account, (error, balance) => {
+            if (error) {
+                reject(error);
+            } else {
+                resolve(balance);
+            }
+        });
+    });
+}
+
+// Calculate the price we paid in ethers for a given transaction receipt.
+async function gasUsedPrice(receipt) {
+    const gasUsed = receipt.gasUsed;
+    const transaction = await new Promise((resolve, reject) => {
+        const web3 = Crowdsale.web3;
+        web3.eth.getTransaction(receipt.transactionHash, (error, result) => {
+            if (error) {
+                reject(error);
+            } else {
+                resolve(result);
+            }
+        });
+    });
+    const gasPrice = transaction.gasPrice;
+    return gasPrice.mul(gasUsed);
+}
+
+// Assert that two big numbers are equal, provide descriptive error message if
+// not.
+function assertEq(got, want, msg) {
```

Comment:
Suggest looking into:
https://github.com/domenic/chai-as-promised
https://www.npmjs.com/package/chai-bignumber

Eg usage:
https://github.com/joincivil/contracts/blob/master/test/utils/chaiconfig.ts
and then
https://github.com/joincivil/contracts/blob/master/test/newsroom.ts#L96

Reply:
I had some issues making Truffle test load extra modules. I'll take another look.